lib/prototyp/bigint.go

```diff
@@ -290,15 +293,34 @@ func ParseNumberString(s string, base int) (int64, bool) {
 		s = strings.TrimPrefix(s, "0X")
 		s = strings.TrimPrefix(s, "0x")
 	default:
+		return BigInt{}, false
+	}
+
+	if neg {
+		s = "-" + s
+	}
+
+	b := big.NewInt(0)
+	b, ok := b.SetString(s, base)
+	if !ok {
+		return BigInt{}, false
+	}
+
+	return BigInt(*b), true
+}
+
+func ParseNumberString(s string, base int) (int64, bool) {
+	bi, ok := ParseBigIntString(s, base)
+	if !ok {
 		return 0, false
 	}
 
-	n, err := strconv.ParseInt(s, base, 64)
-	if err != nil {
```

Comment:
So this was effectively returning `0, false` for all numbers outside of the `int64` range.. which in turn caused `NewBigIntFromString()` to return `BigInt(0)` for all bigger numbers with no error.

And this is exactly [the bug](https://github.com/0xsequence/issue-tracker/issues/4318) you guys were seeing on https://market.champions.io Marketplace.

Great catch, David!